lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -361,10 +361,6 @@ defmodule Kernel.ErrorsTest do
     assert_compile_fail CompileError,
       "nofile:1: unknown key :age for struct Kernel.ErrorsTest.GoodStruct",
       '%#{GoodStruct}{age: 27} = %{}'
-
-    assert_compile_fail CompileError,
-      "nofile:1: unknown key ^field for struct Kernel.ErrorsTest.GoodStruct",
-      '%#{GoodStruct}{^field => 27} = %{}'
```

Comment:
I'm not sure we should test this with `field` being defined or just remove this test. Suggestions are welcome.